Tests/PointFreeTests/AccountTests/CancelTests.swift

```diff
@@ -25,34 +27,43 @@ final class CancelTests: TestCase {
 
   func testCancel() async throws {
     var immediately: Bool?
-    let cancelSubscription = Current.stripe.cancelSubscription
-    Current.stripe.cancelSubscription = {
-      immediately = $1
-      return try await cancelSubscription($0, $1)
-    }
-
-    let conn = connection(from: request(to: .account(.subscription(.cancel)), session: .loggedIn))
+    let expectation = self.expectation(description: "sendEmail")
 
-    await assertSnapshot(matching: conn |> siteMiddleware, as: .ioConn)
+    await withDependencyValues {
+      let cancelSubscription = Current.stripe.cancelSubscription
+      $0.stripe.cancelSubscription = {
+        immediately = $1
+        return try await cancelSubscription($0, $1)
+      }
+      $0.mailgun.sendEmail = { email in
+        expectation.fulfill()
+        XCTAssertEqual(email.subject, "[testing] Your subscription has been canceled")
+        return SendEmailResponse(id: "mail-id", message: "")
+      }
```

Comment:
Here's an example of how I was able to test that the cancellation email is sent when cancelling a subscription.